packages/block-library/src/calendar/index.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Server-side rendering of the `core/calendar` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/calendar` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the block content.
+ */
+function render_block_core_calendar( $attributes ) {
+	global $monthnum, $year, $post;
+	$previous_monthnum;
+	$previous_year;
+
+	// phpcs:disable WordPress.WP.GlobalVariablesOverride.OverrideProhibited
+	if ( isset( $attributes['month'] ) ) {
+		$previous_monthnum = $monthnum;
+		$monthnum          = $attributes['month'];
+	}
+
+	if ( isset( $attributes['year'] ) ) {
+		$previous_year = $year;
+		$year          = $attributes['year'];
+	}
+	// phpcs:enable WordPress.WP.GlobalVariablesOverride.OverrideProhibited
+
+	$custom_class_name = empty( $attributes['className'] ) ? '' : ' ' . $attributes['className'];
+	$align_class_name  = empty( $attributes['align'] ) ? '' : ' ' . "align{$attributes['align']}";
+
+	return sprintf(
+		'<div class="%1$s">%2$s</div>',
+		esc_attr( 'wp-block-calendar' . $custom_class_name . $align_class_name ),
+		get_calendar( true, false )
+	);
+
+	// phpcs:disable WordPress.WP.GlobalVariablesOverride.OverrideProhibited
+	if ( isset( $attributes['month'] ) ) {
```

Comment:
nit: It would be fine to _always_ backup and restore these variables if you want to simplify this method by removing some of the `if`s. 

```php
function render_block_core_calendar( $attributes ) {
	global $monthnum, $year, $post;

	$previous_monthnum = $monthnum;
	$previous_year = $year;

	// ... 

	$monthnum = $previous_monthnum;
	$year = $previous_year;
}
```